test/hotspot/jtreg/runtime/valhalla/inlinetypes/field_layout/NullMarkersTest.java

```diff
@@ -0,0 +1,298 @@
+    Asserts.assertTrue(f5.isFlat());
+    Asserts.assertTrue(f5.hasNullMarker());
+  }
+
+  @ImplicitlyConstructible
+  @LooselyConsistentValue
+  static value class Value4 {
+    int i = 0;
+    byte b = 0;
+  }
+
+  static class Container4 {
+    Value4 val0;
+    Value4 val1;
+  }
+
+  static public void test_4() {
+    Container4 c = new Container4();
+  }
+
+  static void check_4(FieldLayoutAnalyzer fla) {
+    FieldLayoutAnalyzer.ClassLayout cl = fla.getClassLayoutFromName("NullMarkersTest$Container4");
+    FieldLayoutAnalyzer.FieldBlock f0 = cl.getFieldFromName("val0", false);
+    Asserts.assertTrue(f0.isFlat());
+    Asserts.assertTrue(f0.hasNullMarker());
+    FieldLayoutAnalyzer.FieldBlock f1 = cl.getFieldFromName("val1", false);
+    Asserts.assertTrue(f1.isFlat());
+    Asserts.assertTrue(f1.hasNullMarker());
+  }
+
+  @ImplicitlyConstructible
+  @LooselyConsistentValue
+  static value class Value5a {
+    short s = 0;
+    byte b = 0;
+  }
+
+  @ImplicitlyConstructible
+  @LooselyConsistentValue
+  static value class Value5b {
+    @NullRestricted
+    Value5a val0 = new Value5a();
+    @NullRestricted
+    Value5a val1 = new Value5a();
+  }
+
+  static class Container5 {
+    Value5a vala;
+    Value5b valb0;
+    Value5b valb1;
+  }
+
+  static public void test_5() {
+    Container5 c = new Container5();
+  }
+
+  static void check_5(FieldLayoutAnalyzer fla) {
+    FieldLayoutAnalyzer.ClassLayout cl = fla.getClassLayoutFromName("NullMarkersTest$Container5");
+    FieldLayoutAnalyzer.FieldBlock fa = cl.getFieldFromName("vala", false);
+    Asserts.assertTrue(fa.isFlat());
+    Asserts.assertTrue(fa.hasNullMarker());
+    FieldLayoutAnalyzer.FieldBlock fb0 = cl.getFieldFromName("valb0", false);
+    Asserts.assertTrue(fb0.isFlat());
+    Asserts.assertTrue(fb0.hasNullMarker());
+    FieldLayoutAnalyzer.FieldBlock fb1 = cl.getFieldFromName("valb1", false);
+    Asserts.assertTrue(fb1.isFlat());
+    Asserts.assertTrue(fb1.hasNullMarker());
+  }
+
+  static ProcessBuilder exec(String... args) throws Exception {
+    List<String> argsList = new ArrayList<>();
+    Collections.addAll(argsList, "--enable-preview");
+    Collections.addAll(argsList, "-XX:+PrintFieldLayout");
+    Collections.addAll(argsList, "-XX:+EnableNullableFieldFlattening");
+    Collections.addAll(argsList, "-cp", System.getProperty("java.class.path") + ":.");
+    Collections.addAll(argsList, args);
+    return ProcessTools.createTestJavaProcessBuilder(argsList);
+  }
+
+  public static void main(String[] args) throws Exception {
+    // Generate test classes
+    NullMarkersTest fat = new NullMarkersTest();
+
+
+    // Execute the test runner in charge of loading all test classes
+    ProcessBuilder pb = exec("NullMarkersTest$TestRunner");
+    OutputAnalyzer out = new OutputAnalyzer(pb.start());
+
+    Asserts.assertEquals(out.getExitValue(), 0, "Something went wrong while running the tests");
```

Comment:
Speeds up problem solving if the output and error streams for processes is printed...there is the same issue with `PrintFieldLayout` not being product flag

Reply:
The output now contains more information to speed-up investigations. In most cases, the output is too big to be saved by jtreg with a normal configuration, but added information should indicate in which class an error has been detected (last part of the log is saved by jtreg). 
A better solution would be to not print the whole output, but just the layouts of the classes involved in the fail check. But this would require significant work and this patch is already big. I'd prefer to push that in another RFE.